src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs

```diff
@@ -35,119 +35,120 @@
-                                               :workspace-id (:workspace-id props)}]]])}])
+; Render out the import form - Need access to the selected configuration, workspace, and import callback.
+(react/defc ConfigurationImportForm
+  {:render
+   (fn [{:keys [props refs]}]
+     (let [selected-config (:selected-method-config props)
+           selected-conf-name (selected-config "name")
+           selected-conf-namespace (selected-config "namespace")
+           selected-conf-snapshot-id (selected-config "snapshotId")
+           workspace-id (:workspace-id props)]
+       [:div {}
+        (formatted-header "Import Method Configuration")
+        ; Render the form fields:
+        (for [[k v] {"Name" selected-conf-name
+                     "Namespace" selected-conf-namespace
+                     "Snapshot Id" selected-conf-snapshot-id
+                     }] (formatted-label-text k v))
+        (for [[k v] {"Destination Name"
+                     (style/create-text-field {:defaultValue selected-conf-name :ref "destinationName"})
+                     "Destination Namespace"
+                     (style/create-text-field {:defaultValue selected-conf-namespace :ref "destinationNamespace"})}]
+          (formatted-label-textfield k v))
+        ;Render the import button and on-click handler for import
+        [comps/Button
+         {:text "Import Configuration ..."
+          :icon :plus
+          :onClick (fn []
+                     (let [dest-conf-name (.-value (.getDOMNode (@refs "destinationName")))
+                           dest-conf-namespace (.-value (.getDOMNode (@refs "destinationNamespace")))
+                           post-data (create-post-data
+                                       selected-conf-name
+                                       selected-conf-namespace
+                                       selected-conf-snapshot-id
+                                       dest-conf-name
+                                       dest-conf-namespace)]
+                       (utils/ajax-orch
+                         (paths/copy-method-config-to-workspace-path workspace-id)
+                         {:headers {"Content-Type" "application/json"}
+                          :canned-response {:responseText
+                                            (utils/->json-string (create-mock-methodconfs-import))
+                                            :status 200
+                                            :delay-ms (rand-int 2000)}
+                          :method :post
+                          :data (utils/->json-string post-data)
+                          :on-done (fn [{:keys [success? xhr]}]
+                                     (if success?
+                                       ((:on-import props))
+                                       (js/alert (str "Import Error: "  (.-responseText xhr)))))})))}]]))})
 
+; Render out the table component - just need the actual configurations to list and a handler to call when
```

Comment:
I really liked the pattern you used above: bind variables from the properties in a let so it's clear what this component uses. I prefer that over comments like this because it is less likely to become wrong when the code changes.